src/validation.h

```diff
@@ -907,7 +913,7 @@ class ChainstateManager
     //                                  constructor
     //! @param[in] snapshot_blockhash   If given, signify that this chainstate
     //!                                 is based on a snapshot.
-    CChainState& InitializeChainstate(CTxMemPool& mempool, const std::optional<uint256>& snapshot_blockhash = std::nullopt)
+    CChainState& InitializeChainstate(CTxMemPool* mempool, const std::optional<uint256>& snapshot_blockhash = std::nullopt)
```

Comment:
Similarly to [this comment](https://github.com/bitcoin/bitcoin/pull/22415/commits/b17bb4c37d58ca49e877983a632bfcaa493be993#r665701402), can we somehow reflect that if mempool is nullptr, the second argument doesn't make sense?

Reply:
What do you think about making this std::optional<CTxMemPool, std::optional<uint256>>?

In this case, snapshot_blockhash can't be possibly specified without mempool, and that's what I originally suggested.